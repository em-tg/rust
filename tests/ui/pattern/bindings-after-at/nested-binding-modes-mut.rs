//@ check-pass
fn main() {
    let mut is_mut @ not_mut = 42;
    &mut is_mut;
    &mut not_mut;
    //~^ WARNING cannot borrow

    let not_mut @ mut is_mut = 42;
    &mut is_mut;
    &mut not_mut;
    //~^ WARNING cannot borrow
}
