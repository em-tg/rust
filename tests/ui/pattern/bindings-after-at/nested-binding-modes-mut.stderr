warning[E0596]: cannot borrow `not_mut` as mutable, as it is not declared as mutable
  --> $DIR/nested-binding-modes-mut.rs:5:5
   |
LL |     &mut not_mut;
   |     ^^^^^^^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut is_mut @ mut not_mut = 42;
   |                      +++

warning[E0596]: cannot borrow `not_mut` as mutable, as it is not declared as mutable
  --> $DIR/nested-binding-modes-mut.rs:10:5
   |
LL |     &mut not_mut;
   |     ^^^^^^^^^^^^ cannot borrow as mutable
   |
help: consider changing this to be mutable
   |
LL |     let mut not_mut @ mut is_mut = 42;
   |         +++

warning: 2 warnings emitted

For more information about this error, try `rustc --explain E0596`.
